Assets/Scripts/PlayerInteraction/Teleport.cs

```diff
@@ -0,0 +1,113 @@
+        [Tooltip("The Layer Mask of layers that are valid destinations for a teleport. Players will only be able to teleport to objects on these layers.")]
+        public LayerMask validLayers;
+        [Tooltip("The trigger used to determine when the player is attempting to teleport.")]
+        public Button teleportButton;
+        [Tooltip("How far the trigger must be pulled before a raycast occurs.")]
+        public float teleportRange = 500f;
+        [Tooltip("The line renderer to use to visualize the raycast.")]
+        public LineRenderer lineRenderer;
+
+        // Whether the current teleportation attempt is valid (The raycast hit a layer in the 'validLayers' Layer Mask)
+        private bool valid;
+        // The coordinate the player will teleport to when the trigger is released (The location that the raycast hit)
+        private Vector3 destination;
+        // Input manager to check for button press
+        private InputManager inputManager;
+
+        void Start()
+        {
+            // Get an instance of an input maanger
+            inputManager = InputManager.Instance;
+            // Set default values
+            valid = false;
+            destination = Vector3.zero;
+            // Disable the line renderer
+            lineRenderer.enabled = false;
+        }
+
+        void Update()
+        {
+            // Check if the trigger has been indented
+            if (inputManager.IsButtonPressed(teleportButton))
+            {
+                // Enable the line renderer
+                lineRenderer.enabled = true;
+                // The first point on the line renderer will be the raycast origin
+                var points = new Vector3[2];
+                points[0] = rayOrigin.position;
+
+                // Send a raycast out from the rayOrigin
+                RaycastHit hit;
+                if (Physics.Raycast(rayOrigin.position, rayOrigin.rotation * Vector3.forward, out hit, teleportRange, layerMask))
+                {
+                    // Determine if the hit game object's layer is one of the valid layers. (from https://answers.unity.com/questions/50279/check-if-layer-is-in-layermask.html)
+                    valid = validLayers == (validLayers | (1 << hit.transform.gameObject.layer));
+                    // If the layer is valid, save the hit location
+                    if (valid)
+                    {
+                        destination = hit.point;
+                    }
+                    // Set the second point on the renderer to the raycast's hit point
+                    points[1] = hit.point;
+                }
+                else
+                {
+                    // If raycast does not hit, then the teleport cannot be valid
+                    valid = false;
+                    // Set the second point on the renderer to the max distance in front of the origin
+                    points[1] = rayOrigin.position + rayOrigin.rotation * Vector3.forward * teleportRange;
```

Comment:
Same here. Doing `rayOrigin.forward` should give you the same result as `rayOrigin.rotation * Vector3.forward`